zdm/iteration.py

```diff
@@ -194,19 +215,41 @@ def update_grid(grid,pset):
 	alpha=pset[2]
 	gamma=pset[3]
 	sfrn=pset[4]
-	
+	H0value=pset[8]
+
 	# assumes all parameters following this are passed to DM distribution
 	smear_mean=pset[5]
 	smear_sigma=pset[6]
-	oldlEmin=np.log10(grid.Emin)
-	oldlEmax=np.log10(grid.Emax)
+
+	#taking all the old grid parameters
+    oldmask=grid.smear
+    oldsmeargrid=grid.smear_grid
+    
+	oldlEmin=np.log10(float(grid.Emin))
+	oldlEmax=np.log10(float(grid.Emax))
 	oldalpha=grid.alpha
 	oldgamma=grid.gamma
 	oldsfrn=grid.sfr_n
 	oldsmean=grid.smear_mean
 	oldssigma=grid.smear_sigma
+	oldH0value=grid.H0
 	new_sfr_smear=False #do not need to calculate new product of sfr and smearing
 	
+	##check if H0 value changed
+    
+    if oldH0value != pset[8]:
+        cos.set_cosmology(H0=H0value)
+        zDMgrid, zvals,dmvals,H0=misc_functions.get_zdm_grid(H0=H0value,new=True,plot=False,method='analytic')
+        grid.pass_grid(zDMgrid,zvals,dmvals,H0)
+        grid.smear_mean=oldsmean
+        grid.smear_sigma=oldssigma
+        grid.smear_dm(oldmask,oldsmean,oldssigma)
+        grid.calc_thresholds(survey.meta['THRESH'],survey.efficiencies,alpha=oldalpha, weights=survey.wplist)
```

Comment:
The weights from survey.wlist shouldn't change I don't think. Thresholds will, yes, but can you recycle the wplist from the grid being passed? If the grid does not store survey.efficiencies or survey.wplist then this is fine I guess. Just wondering. Clearly you have to call "calc_thresholds" again. E.g. re-use "weights=grid.eff_weights" should work. Same with efficiencies=grid.eff_table. I don't think efficiencies or eff_weights changes with H0?

Reply:
No, none of them change! I am indeed reusing the `wplist, efficiencies`, etc. However they are stored in the `survey` object, **not** the `grid `object. This is actually why `update_grid()` is taking another parameter which is the `survey` itself.